h2/connection.py

```diff
@@ -425,6 +427,7 @@ def _open_streams(self, remainder):
                 to_delete.append(stream_id)
 
         for stream_id in to_delete:
+            self._closed_streams[stream_id] = self.streams[stream_id].closed_by
             del self.streams[stream_id]
```

Comment:
We can avoid doing two dict lookups here by changing the first line to `stream = self.streams.pop(stream_id)` and then do `self._closed_streams[stream_id] = stream.closed_by`. Moderately more efficient.